src/hooks/useSelectedResult.ts

```diff
@@ -0,0 +1,27 @@
+import { useState, useEffect } from "react";
+import yelp from "../api/yelp";
+
+export interface Result {
+  id: string;
+  name: string;
+  image_url: string;
+  review_count: number;
+  rating: number;
+  photos: string[];
+  price: string;
+}
+
+export default (id: string) => {
+  const [result, setResult] = useState<Result | null>(null);
+
+  const getResult = async (id: string): Promise<void> => {
```

Comment:
Since `getResult` is only used inside the `useEffect` callback, it should be moved into it. I'll add a code suggestion so you see what I mean.